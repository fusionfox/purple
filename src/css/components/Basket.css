.Basket {
  margin: 10px;
}

.Basket-title {
  font-weight: bold
}

.Basket-list {
  padding: 0;
}

.Basket-button {
  padding: 20px 40px;
  font-size: 15px;
}
