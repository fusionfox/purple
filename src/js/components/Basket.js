import React from 'react'
import '../../css/components/Basket.css'

const Basket = props => {
  const products = props.products.map(product => {
    return <li key={product.id} className='Basket-product'>{product.name}</li>
  })

  const EmptyBasketMessage = props => {
    if (products.length === 0) {
      return <p>Basket is empty</p>
    } else {
      return null
    }
  }

  return (
    <figure className='Basket'>
      <figcaption className='Basket-title'>Basket</figcaption>
      <ul className='Basket-list'>
        {products}
      </ul>
      <EmptyBasketMessage />
      <button className='Basket-button' onClick={props.handleCheckout}>Checkout</button>
    </figure>
  )
}

export default Basket
