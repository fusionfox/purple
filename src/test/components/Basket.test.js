import React from 'react'
import { BrowserRouter as Router } from 'react-router-dom'
import renderer from 'react-test-renderer'
import Basket from '../../js/components/Basket'

const products = [
  {id: 'pr1', name: 'Number One'},
  {id: 'pr2', name: 'Number Two'}
]

describe('Basket component', () => {
  it('Renders a list of products with the title "Basket" and a checkout button', () => {
    const component = renderer.create(
      <Router>
        <Basket products={products} />
      </Router>
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('Renders a message if given no products', () => {
    const component = renderer.create(
      <Router>
        <Basket products={[]} />
      </Router>
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
