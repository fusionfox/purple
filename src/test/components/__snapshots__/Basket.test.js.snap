// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basket component Renders a list of products with the title "Basket" and a checkout button 1`] = `
<figure
  className="Basket"
>
  <figcaption
    className="Basket-title"
  >
    Basket
  </figcaption>
  <ul
    className="Basket-list"
  >
    <li
      className="Basket-product"
    >
      Number One
    </li>
    <li
      className="Basket-product"
    >
      Number Two
    </li>
  </ul>
  <button
    className="Basket-button"
    onClick={[Function]}
  >
    Checkout
  </button>
</figure>
`;

exports[`Basket component Renders a message if given no products 1`] = `
<figure
  className="Basket"
>
  <figcaption
    className="Basket-title"
  >
    Basket
  </figcaption>
  <ul
    className="Basket-list"
  />
  <p>
    Basket is empty
  </p>
  <button
    className="Basket-button"
    onClick={[Function]}
  >
    Checkout
  </button>
</figure>
`;
